### PR TITLE
Omit 'schema' as part of 'Table.{create,update}' when 'view_query` is set

### LINKED ISSUE #1701: schema built when creating a view in bigquery
Calling create on a table with view_query set fails, as _build_resource sets the schema. Running

```python
new_table = dataset.table('test_view')
table.view_query = query_string
new_table.create()
```

gives the following error:

```
BadRequest: 400 {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "invalid",
    "message": "Schema field shouldn't be used as input with a view"
   }
  ],
  "code": 400,
  "message": "Schema field shouldn't be used as input with a view"
 }
}
```

One fix that works is only setting `schema` if `view_query` is `None` in `_build_resource` in [`gcloud/bigquery/table.py`][1], e.g.:

```python
    def _build_resource(self):
        """Generate a resource for ``create`` or ``update``."""
        resource = {
            'tableReference': {
                'projectId': self._dataset.project,
                'datasetId': self._dataset.name,
                'tableId': self.name},
        }
        if self.description is not None:
            resource['description'] = self.description

        if self.expires is not None:
            value = _millis_from_datetime(self.expires)
            resource['expirationTime'] = value

        if self.friendly_name is not None:
            resource['friendlyName'] = self.friendly_name

        if self.location is not None:
            resource['location'] = self.location

        if self.view_query is not None:
            view = resource['view'] = {}
            view['query'] = self.view_query
        else:
            resource['schema'] = {'fields': _build_schema_resource(self._schema)}

        return resource
```

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/188515f192e3823ce21a47df2d4742995d08f109/gcloud/bigquery/table.py#L397

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1701.

### CODE REVIEW

File: gcloud/bigquery/table.py [hunk 1]
```diff
@@ -419,6 +419,8 @@ def _build_resource(self):
         if self.view_query is not None:
             view = resource['view'] = {}
             view['query'] = self.view_query
+            # https://github.com/GoogleCloudPlatform/gcloud-python/issues/1701
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: Why is this comment necessary?

**Comment by dhermes** [comment]
Lint errors:

```
************* Module gcloud.bigquery.test_table
I:1400, 0: Locally disabling too-many-statements (R0915) (locally-disabled)
I:1602, 0: Locally disabling unused-argument (W0613) (locally-disabled)
W:800, 8: Unused variable 'full_name' (unused-variable)
W:801, 8: Unused variable 'age' (unused-variable)
```

File: gcloud/bigquery/table.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigquery/table.py]: Something mimicking the error message "Schema field shouldn't be used as input with a view" would be better

Comment (tseaver) [inline gcloud/bigquery/table.py]: The comment explains why one is mutating an apparently-unrleated bit of the resource when passing the `{'view': {'query': QUERY_STRING}}` element.

File: gcloud/bigquery/table.py [hunk 2]
```diff
@@ -419,6 +419,8 @@ def _build_resource(self):
         if self.view_query is not None:
             view = resource['view'] = {}
             view['query'] = self.view_query
+            # Back-end rejects 'schema' when passing a query.
+            del resource['schema']
```
Comment (dhermes) [inline gcloud/bigquery/table.py]: @tseaver Why not

```python
if self.view_query is not None:
     view = resource['view'] = {}
     view['query'] = self.view_query
else:
     # Schema can only be used on non-queries.
     resource['schema'] = {'fields': _build_schema_resource(self._schema)}
```

and then just drop out the `schema` from `resource` above. Seems a bit silly to add and `del` when we could do this and avoid instead.

**Comment by tseaver** [comment]
@dhermes Any remaining issue?

**Comment by dhermes** [comment]
LGTM. Squash if you have the energy for it.

**Pull request merged**